### PR TITLE
Fixes Save Button to Follow Keyboard

### PR DESCRIPTION
**Pull request by redreceipt**
## DESCRIPTION

Kapture 2019-06-16 at 17 33 10

### What does this PR do, or why is it needed?

Makes the save button on the Personal Details form follow the keyboard

### What design trade-offs/decisions were made?

Hard codes a vertical offset number to account for the nav header

### How do I test this PR?

- go to edit your Personal details

### What automated tests did you write?

None

## TODO:

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [x] Closes [#766]
- [x] No new warnings in tests, in storybook, and in-app
- [x] Upload GIF(s) of iOS and Android
- [x] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Comment by vinnyjth** [comment]
image

**Comment by vinnyjth** [comment]
We're looking pretty good @redreceipt. I have a comment about the `verticalOffset` though, which might be related to why we don't have a vertical offset on Android.

**Comment by redreceipt** [comment]
> We're looking pretty good @redreceipt. I have a comment about the `verticalOffset` though, which might be related to why we don't have a vertical offset on Android.

???

File: packages/apolloschurchapp/src/user-settings/PersonalDetails.js [hunk 1]
```diff
@@ -43,8 +43,13 @@ class PersonalDetails extends PureComponent {
   };
 
   renderForm = (props) => (
-    <FlexedView>
-      <TableView>
+    // have to add the offset to account for react-navigation header
+    <KeyboardAvoidingView
+      behavior={'padding'}
+      style={StyleSheet.absoluteFill}
+      keyboardVerticalOffset={64}
```
Comment (vinnyjth) [inline packages/apolloschurchapp/src/user-settings/PersonalDetails.js]: It bums me out that we have to hardcode this. Have you looked at solutions like `Header.HEIGHT` from ReactNavigation? Based on some research I did, I'm also not sure 64 is the header height, is that actually the `tabBar` height? @conrad-vanl - any ideas here? It might be a better idea to make sure the button padding is working better.

Comment (redreceipt) [reply packages/apolloschurchapp/src/user-settings/PersonalDetails.js]: `Header.HEIGHT` helped a little but it seems I also need to take into account the Android title bar for some reason, any ideas?

Screen Shot 2019-06-18 at 11 11 52 AM

**Comment by redreceipt** [comment]
Android `Dimensions.height` doesn't include status bar, this is the issue I think. Trying to find a way to add that in...

Screen Shot 2019-06-19 at 11 46 37 AM

**Comment by redreceipt** [comment]
@vinnyjth ok this one's ready, would love @burkeshartsis thoughts too.
